x-pack/plugins/ingest_manager/server/services/agent_policy.ts

```diff
@@ -119,6 +121,30 @@ class AgentPolicyService {
     return { id: newSo.id, ...newSo.attributes };
   }
 
+  public async requireUniqueName(
+    soClient: SavedObjectsClientContract,
+    { name, namespace }: Pick<NewAgentPolicy, 'name' | 'namespace'>
+  ) {
+    const results = await soClient.find<AgentPolicySOAttributes>({
+      type: SAVED_OBJECT_TYPE,
+      searchFields: ['namespace', 'name'],
+      search: `${namespace} + ${escapeSearchQueryPhrase(name)}`,
+    });
+
+    if (results.total) {
+      const policies = results.saved_objects;
+      const isSinglePolicy = policies.length === 1;
+      const policyList = isSinglePolicy ? policies[0].id : policies.map(({ id }) => id).join(',');
+      const existClause = isSinglePolicy
+        ? `Agent Policy '${policyList}' already exists`
+        : `Agent Policies '${policyList}' already exist`;
+
+      throw new AgentPolicyNameExistsError(
+        `${existClause} in '${namespace}' namespace with name '${name}'`
```

Comment:
I'm confused as to where the uniqueness-per-namespace requirement came from, this is the first I've heard of it and I don't see it discussed in the original issue. `AgentPolicy.namespace` =/= Kibana space. I understood the requirement for this to be `AgentPolicy.name` uniqueness across Fleet, regardless of the namespace string.

Reply:
Yes, the requirement is the name to be unique per "installation" we don't support space at the moment.

Reply:
@jen-huang I might misunderstood the conversation during the sync.

Should https://github.com/elastic/kibana/pull/79201/files#diff-bf1476574510be26b7814652c0480832R228 return a `409` instead?

If so, I'd like to do that ASAP in a follow-up PR since we're so close to FF (and CI so slow) & I believe this is a low impact bug to be in main for a few hours.

Reply:
yes, that should be 409. as this is a bug fix, this can go in after FF.

Reply:
sound good.

Reply:
I created #79903 to track the work needed for a follow up PR